src/map-marker/fixed-marker.js

```diff
@@ -107,24 +107,35 @@ const FixedMarker = ({
     size !== PINHEAD_SIZES_SHAPES.xxSmallCircle &&
     size !== PINHEAD_SIZES_SHAPES.xxSmallSquare;
 
-  const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
+  const [Root, rootProps] = getOverrides(overrides.Root, StyledFixedMarkerRoot);
   const [
     FixedMarkerDragContainer,
     fixedMarkerDragContainerProps,
   ] = getOverrides(overrides.DragContainer, StyledFixedMarkerDragContainer);
 
   const renderNeedle =
     needle !== NEEDLE_SIZES.none &&
-    size !== PINHEAD_SIZES_SHAPES.xxSmallCircle &&
-    size !== PINHEAD_SIZES_SHAPES.xxSmallSquare;
-  if (!renderNeedle) {
-    console.warn(
-      `Needles cannot be rendered with ${PINHEAD_SIZES_SHAPES.xxSmallCircle} or ${PINHEAD_SIZES_SHAPES.xxSmallSquare} pin heads`,
-    );
+    ![
+      PINHEAD_SIZES_SHAPES.xxSmallCircle,
+      PINHEAD_SIZES_SHAPES.xxSmallSquare,
+    ].includes(size);
+
+  if (__DEV__) {
+    if (
+      needle !== NEEDLE_SIZES.none &&
+      [
```

Comment:
Note, this condition is slightly different from the one above for the `renderNeedle` so that it doesn't show the warning when the `needle` is `NEEDLE_SIZES.none`